### ISSUE TITLE
/tmp does not get mounted to tmpfs correctly when using latest AL2 base

### ISSUE THREAD

**Issue by chrisdleech-pol** [ansible-lockdown/AMAZON2-CIS#21]
**Question**
The following section does not seem to work as intended, presuming my understanding is correct

https://github.com/ansible-lockdown/AMAZON2-CIS/blob/devel/tasks/section_1/cis_1.1.x.yml#L3

This section is supposed to mount /tmp to tmpfs, however the behaviour is incorrect, it mounts /tmp to the device name

Following a build, fstab is configured like this, and mounting /tmp to the root volume device, not tmpfs:

```
sh-4.2$ cat /etc/fstab
#
UUID=9697d457-123c-483b-a84c-efd437479b18     /           xfs    defaults,noatime  1   1
/dev/nvme0n1p1 /tmp xfs defaults,nodev,noexec,nosuid 0 0
tmpfs /dev/shm tmpfs defaults,noexec,nodev,nosuid,seclabel 0 0
```

and the tmp.mount service is configured like:

```
sh-4.2$ systemctl status tmp.mount
● tmp.mount - /tmp
   Loaded: loaded (/etc/fstab; enabled; vendor preset: disabled)
   Active: active (mounted) since Fri 2023-07-28 12:25:22 UTC; 4min 59s ago
    Where: /tmp
     What: /dev/nvme0n1p1
     Docs: man:fstab(5)
           man:systemd-fstab-generator(8)
```

We are using the latest available base version of Amazon Linux 2

To workaround this we have had to strip out this section and replace it with one which simply mounts /tmp to tempfs. 

This actually causes issues with running instances, as we see them periodically fail health checks and become unresponsive. 

I would like to know if this issue is specific to our configuration, or is this a known issue in the role.

**Comment by uk-bolly** [comment]
hi @chrisdleech-pol 

Many apologies for the time this has taken, thank you for your patience. This is actually an issue and not a question so should have been addressed.
I have now added a fix for this to choose between the tmp svc and fstab. It is tested and works as expected and should resolve the issue.
I am currently just creating the final PR release for v2.0 - v3.0 has already been released to subscribers(which is a complete rewrite)

Very kindest regards and thank you again for your patience.

uk-bolly
